clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -2061,7 +2062,18 @@ protected long getTopicPartitionEndOffSet(String kafkaUrl, PubSubTopic pubSubTop
     if (offsetFromConsumer >= 0) {
       return offsetFromConsumer;
     }
-    return getTopicManager(kafkaUrl).getLatestOffsetCached(pubSubTopic, partition);
+    try {
+      return RetryUtils.executeWithMaxAttempt(() -> {
+        long offset = getTopicManager(kafkaUrl).getLatestOffsetCachedNonBlocking(pubSubTopic, partition);
+        if (offset == -1) {
+          throw new VeniceException("Found latest offset -1");
+        }
+        return offset;
+      }, 5, Duration.ofSeconds(1), Collections.singletonList(VeniceException.class));
+    } catch (Exception e) {
+      LOGGER.error("Could not find latest offset for {} even after 5 retries", pubSubTopic.getName());
+      return -1;
```

Comment:
I am wondering if return -1 is expected. If the VT exists, RT does not exist, we return -1 and if the latest processed offset is initialized to 0 (is it true?) then the lag is 1, and it will not block the partition to go online?

Reply:
In RTS code caller `measureRTOffsetLagForSingleRegion`, it does this

`if (lastOffsetInRealTimeTopic < 0) {
      if (!REDUNDANT_LOGGING_FILTER.isRedundantException("Got a negative lastOffsetInRealTimeTopic")) {
        LOGGER.warn(
            "Unexpected! Got a negative lastOffsetInRealTimeTopic ({})! Will return Long.MAX_VALUE ({}) as the lag for replica: {}.",
            lastOffsetInRealTimeTopic,
            Long.MAX_VALUE,
            pcs.getReplicaId(),
            new VeniceException("Exception not thrown, just for logging purposes."));
      }
      return Long.MAX_VALUE;
    }

Reply:
Synced offline - I think it is ok for now.